src/3d/framegraph/qgsdepthrenderview.cpp

```diff
@@ -0,0 +1,130 @@
+}
+
+Qt3DRender::QRenderTarget *QgsDepthRenderView::buildTextures( QSize size )
+{
+  // depth buffer render to copy pass
+  // Create a render target output for rendering color.
+  Qt3DRender::QRenderTargetOutput *colorOutput = new Qt3DRender::QRenderTargetOutput;
+  colorOutput->setAttachmentPoint( Qt3DRender::QRenderTargetOutput::Color0 );
+
+  // Create a texture to render into.
+  mColorTexture = new Qt3DRender::QTexture2D( colorOutput );
+  mColorTexture->setSize( size.width(), size.height() );
+  mColorTexture->setFormat( Qt3DRender::QAbstractTexture::RGB8_UNorm );
+  mColorTexture->setMinificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mColorTexture->setMagnificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mColorTexture->setObjectName( mViewName + "::mColorTexture" );
+
+  // Hook the texture up to our output, and the output up to this object.
+  colorOutput->setTexture( mColorTexture );
+
+  Qt3DRender::QRenderTargetOutput *depthOutput = new Qt3DRender::QRenderTargetOutput;
+
+  depthOutput->setAttachmentPoint( Qt3DRender::QRenderTargetOutput::Depth );
+  mDepthTexture = new Qt3DRender::QTexture2D( depthOutput );
+  mDepthTexture->setSize( size.width(), size.height() );
+  mDepthTexture->setFormat( Qt3DRender::QAbstractTexture::DepthFormat );
+  mDepthTexture->setMinificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mDepthTexture->setMagnificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mDepthTexture->setComparisonFunction( Qt3DRender::QAbstractTexture::CompareLessEqual );
+  mDepthTexture->setComparisonMode( Qt3DRender::QAbstractTexture::CompareRefToTexture );
+  mDepthTexture->setObjectName( mViewName + "::mDepthTexture" );
+
+  depthOutput->setTexture( mDepthTexture );
+
+  Qt3DRender::QRenderTarget *renderTarget = new Qt3DRender::QRenderTarget;
+  renderTarget->setObjectName( mViewName + "::RenderTarget" );
+  renderTarget->addOutput( colorOutput );
+  renderTarget->addOutput( depthOutput );
+
+  return renderTarget;
+}
+
+void QgsDepthRenderView::buildRenderPass( QSize size,                                  //
+                                          Qt3DRender::QTexture2D *forwardDepthTexture, //
+                                          Qt3DCore::QEntity *rootSceneEntity )
+{
+  // depth buffer render to copy pass
+
+  Qt3DRender::QCameraSelector *cameraSelector = new Qt3DRender::QCameraSelector( mRendererEnabler );
+  cameraSelector->setObjectName( mViewName + "::CameraSelector" );
+  cameraSelector->setCamera( mMainCamera );
+
+  Qt3DRender::QRenderStateSet *stateSet = new Qt3DRender::QRenderStateSet( cameraSelector );
+
+  Qt3DRender::QDepthTest *depthTest = new Qt3DRender::QDepthTest;
+  depthTest->setDepthFunction( Qt3DRender::QDepthTest::Always );
+  Qt3DRender::QCullFace *cullFace = new Qt3DRender::QCullFace;
+  cullFace->setMode( Qt3DRender::QCullFace::NoCulling );
+
+  stateSet->addRenderState( depthTest );
+  stateSet->addRenderState( cullFace );
```

Comment:
QCameraSelector actually should not be needed - there's no camera used in the vertex shader for the depth entity (and the fact we use main camera is just confusing)

QRenderStateSet also should not be needed - we don't need to worry about depth tests or culling - it's just one quad we render